app/view_models/Amount.scala

```diff
@@ -46,6 +46,8 @@ case class Amount(amount: BigDecimal, currency: String) {
   def toHundredthsString = format(2, BigDecimal.RoundingMode.DOWN, amount = amount * 100)
 
   def unary_-(): Amount = copy(amount = -this.amount)
+
+  def noLessThanZero: Amount = copy(amount = if(this.amount > 0) this.amount else 0)
```

Comment:
Isn't this.amount a double? How come it's being compared to a BigDecimal?

Reply:
this.amount is a BigDecimal, not a double.